docs/installation/server-config/security.rst

```diff
@@ -1,4 +1,227 @@
+    will create sink topics whenever you are doing a persistent CTAS/CTAS query.
+
+Change-log and repartition topics
+    Internally KSQL uses repartition topics and changelog topics for selected operations. KSQL users minimally require a
+    ``DESCRIBE``, ``READ``, and ``WRITE`` permissions for each changelog and repartition ``TOPIC``.
+
+    By default, KSQL will attempt to create any repartition or changelog topics that do not exist. To allow this, the KSQL
+    user requires ``CREATE`` permissions on the ``CLUSTER``. Alternatively, topics can be created manually before running KSQL.
+    To determine the list of topics and their required configuration, for example, partition count, replication factor,
+    retention policy, etc, you can run KSQL on open cluster first.
+
+    KSQL requires repartition topics when using either ``PARTITION BY``, or using ``GROUP BY`` on none-key values. KSQL
+    requires changelog topics for any ``CREATE TABLE x AS`` statements.
+
+    All changelog and repartition topics are prefixed with ``_confluent-ksql-<value of ksql.service.id property>_query_<query id>_``
+    where ``ksql.service.id`` defaults to ``ksql_``.
+
+Consumer groups
+    The set of consumer groups that KSQL requires depends on the queries that are being executed. The easiest way to
+    determine the list of consumer groups is to first run the queries on an unsecured or open Kafka cluster and list the
+    groups created. For more information about how to list groups, see
+    `Managing Consumer Groups <http://kafka.apache.org/documentation.html#basic_ops_consumer_group>`__.
+
+    Consumer group names are formatted like ``<value of ksql.service.id property>_query_<query id>``, where the default
+    of ``ksql.service.id`` is ``ksql_``.
+
+    KSQL users require a minimum of ``DESCRIBE`` and ``READ`` permissions for ``GROUP``.
+
+.. _security-config-ref:
+
+Security Configuration Reference
+--------------------------------
+
+------------------------------------------
+Global SSL and SASL configuration for KSQL
+------------------------------------------
+
+.. code:: bash
+
+    ssl.truststore.location=/etc/kafka/secrets/kafka.client.truststore.jks
+    ssl.truststore.password=confluent
+    ssl.keystore.location=/etc/kafka/secrets/kafka.client.keystore.jks
+    ssl.keystore.password=confluent
+    ssl.key.password=confluent
+    ssl.endpoint.identification.algorithm=HTTPS
+    security.protocol=SASL_SSL
+    sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="client" password="client-secret";
+    sasl.mechanism=PLAIN
+
+.. tip:: You must create the truststore. For more information, see the `Apache Encryption and Authentication using SSL
+         documentation <http://kafka.apache.org/documentation.html#security_ssl>`_.
+
+--------------------------------------------------------------------------------
+Producer Confluent Monitoring Interceptors for Control Center Streams Monitoring
```

Comment:
This section needs a preamble.  Where should this be config set? Maybe:

> If you are an enterprise customer, who is making use of the `MonitoringProducerInterceptor` to integrate KSQL with C3 and wish to use KSQL connecting to an Apache Kafka cluster using SASL and SSL, then you will need to add properties similar to the following to the KSQL server configuration file.

There is also this docs page that talks about this:
https://github.com/confluentinc/docs/blob/e2df9229b77d10be9b7be5ba2648cf093079300d/kafka/security/auth_sasl_plain_interceptors_config.rst

I've also looked at the `MonitoringProducerInterceptor` code.  Unfortunately, it looks like you do need to duplicate the trust store / key store info again you your config file with the `confluent.monitoring.interceptor` prefix, which isn't nice.

I've created #1002 to track.

Reply:
@big-andy-coates i rolled this example up into the main configuration topic above.